runtime/caml/misc.h

```diff
@@ -39,6 +39,10 @@
 #define __has_attribute(x) 0
 #endif
 
+#ifndef __has_builtin
+#define __has_builtin(x) 0
```

Comment:
I'm not opposed to this myself, but strictly we're not supposed to do this, because `__` identifier prefixes are reserved to the C implementation. This may be why `Caml_has_builtin` existed previously.

Reply:
I agree, but both [clang](https://clang.llvm.org/docs/LanguageExtensions.html#feature-checking-macros) and [GCC](https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005fattribute.html) document this snippet. Linux and libzstd use this pattern too, which is why I think that problems are unlikely to occur.